src/e2e/java/teammates/e2e/cases/lnp/InstructorUpdateLNPTest.java

```diff
@@ -0,0 +1,367 @@
+
+                FeedbackSessionAttributes session = FeedbackSessionAttributes
+                        .builder(FEEDBACK_SESSION_NAME, COURSE_ID)
+                        .withCreatorEmail(INSTRUCTOR_EMAIL)
+                        .withStartTime(Instant.now())
+                        .withEndTime(Instant.now().plusSeconds(500))
+                        .withSessionVisibleFromTime(Instant.now())
+                        .withResultsVisibleFromTime(Instant.now())
+                        .build();
+
+                feedbackSessions.put(FEEDBACK_SESSION_NAME, session);
+
+                return feedbackSessions;
+            }
+
+            @Override
+            protected Map<String, FeedbackQuestionAttributes> generateFeedbackQuestions() {
+                ArrayList<FeedbackParticipantType> showResponses = new ArrayList<>();
+                showResponses.add(FeedbackParticipantType.RECEIVER);
+                showResponses.add(FeedbackParticipantType.INSTRUCTORS);
+
+                ArrayList<FeedbackParticipantType> showGiverName = new ArrayList<>();
+                showGiverName.add(FeedbackParticipantType.INSTRUCTORS);
+
+                ArrayList<FeedbackParticipantType> showRecepientName = new ArrayList<>();
+                showRecepientName.add(FeedbackParticipantType.INSTRUCTORS);
+
+                Map<String, FeedbackQuestionAttributes> feedbackQuestions = new LinkedHashMap<>();
+                FeedbackQuestionDetails details = new FeedbackTextQuestionDetails(FEEDBACK_QUESTION_TEXT);
+
+                feedbackQuestions.put(FEEDBACK_QUESTION_ID,
+                        FeedbackQuestionAttributes.builder()
+                                .withFeedbackSessionName(FEEDBACK_SESSION_NAME)
+                                .withQuestionDescription(FEEDBACK_QUESTION_TEXT)
+                                .withCourseId(COURSE_ID)
+                                .withQuestionDetails(details)
+                                .withQuestionNumber(1)
+                                .withGiverType(FeedbackParticipantType.INSTRUCTORS)
+                                .withRecipientType(FeedbackParticipantType.STUDENTS)
+                                .withShowResponsesTo(showResponses)
+                                .withShowGiverNameTo(showGiverName)
+                                .withShowRecipientNameTo(showRecepientName)
+                                .build()
+                );
+
+                return feedbackQuestions;
+            }
+
+            @Override
+            protected Map<String, FeedbackResponseAttributes> generateFeedbackResponses() {
+                Map<String, FeedbackResponseAttributes> feedbackResponses = new HashMap<>();
+                FeedbackTextResponseDetails details =
+                        new FeedbackTextResponseDetails(FEEDBACK_RESPONSE_ID);
+
+                feedbackResponses.put(FEEDBACK_RESPONSE_ID,
+                        FeedbackResponseAttributes.builder(FEEDBACK_QUESTION_ID,
```

Comment:
Here is the mistake.

The feedback question ID is randomly generated and thus cannot be guessed beforehand. To circumvent that problem, the `DataBundleLogic` is designed such that it will read the existing feedback question by using the course ID, feedback session name, and question number which is hacked (in a way) to be the feedback question ID field in the JSON data.

So here the value you should use is `"1"`, as you have only added one feedback question.

Reply:
Awesome! This is exactly the cause ;)